lib/core/shell.rb

```diff
@@ -1,6 +1,10 @@
 # frozen_string_literal: true
 
 class Shell
```

Comment:
Should we have 2 class `Shell` in this project?

Reply:
You mean because of Ruby's [`Shell`](https://ruby-doc.org/stdlib-2.5.1/libdoc/shell/rdoc/Shell.html)? We could rename it, maybe to `ThorShell`?